--- filehighlight.ini.orig	2016-03-31 08:51:32.172955928 +0200
+++ filehighlight.ini	2016-03-31 08:51:24.630050227 +0200
@@ -28,7 +28,7 @@
     regexp=(^#.*|.*~$)
 
 [archive]
-    extensions=7z;Z;ace;arc;arj;ark;bz2;cab;gz;lha;lz;lz4;lzh;lzma;rar;rpm;tar;tbz;tbz2;tgz;tlz;xz;zip;zoo
+    extensions=7z;Z;ace;arc;arj;ark;bz2;cab;gz;lha;lz;lz4;lzh;lzma;rar;rpm;tar;tbz;tbz2;tgz;tlz;txz;xz;zip;zoo
 
 [doc]
     extensions=chm;css;ctl;diz;doc;docm;docx;dtd;htm;html;letter;lsm;mail;man;me;msg;nroff;pdf;po;pptm;pptx;ps;rtf;sgml;shtml;tex;text;txt;xlsm;xlsx;xml;xsd;xslt
